sky/provision/paperspace/instance.py

```
@@ -0,0 +1,325 @@
+                    f'{len(exist_instances)} nodes, no need to start more.')
+        return common.ProvisionRecord(
+            provider_name='paperspace',
+            cluster_name=cluster_name_on_cloud,
+            region=region,
+            zone=None,
+            head_instance_id=head_instance_id,
+            resumed_instance_ids=list(newly_started_instances.keys()),
+            created_instance_ids=[],
+        )
+
+    created_instance_ids = []
+    for _ in range(to_start_count):
+        node_type = 'head' if head_instance_id is None else 'worker'
+        try:
+            instance_id = client.launch(
+                name=f'{cluster_name_on_cloud}-{node_type}',
+                instance_type=config.node_config['InstanceType'],
+                network_id=config.node_config['NetworkId'],
+                region=region,
+                disk_size=config.node_config['DiskSize'],
+            )['data']['id']
+        except Exception as e:  # pylint: disable=broad-except
+            logger.warning(f'run_instances error: {e}')
+            raise e
+        logger.info(f'Launched instance {instance_id}.')
+        created_instance_ids.append(instance_id)
+        if head_instance_id is None:
+            head_instance_id = instance_id
+
+    # Wait for instances to be ready.
+    for _ in range(MAX_POLLS_FOR_UP_OR_STOP):
+        instances = _filter_instances(cluster_name_on_cloud, ['ready'])
+        logger.info('Waiting for instances to be ready: '
+                    f'({len(instances)}/{config.count}).')
+        if len(instances) == config.count:
+            break
+
+        stopped_instances = _filter_instances(cluster_name_on_cloud, ['off'])
+        # Paperspace will turn off the machine if there is a capacity issue
+        if stopped_instances:
+            msg = ('run_instances: Found a stopped instance during creation '
+                   'likely due to capacity issue.')
+            logger.warning(msg)
+            raise RuntimeError(msg)
```

Ahh, sorry for the back and forth, but when I test more with this, it seems paperspace will have the instance in `off` state for a short period before it starts `starting` the instance, even if the capacity is available. With that, the current handling for failing early can cause the provisioning fail. Let's get rid of this optimization for now.
```suggestion
```